main.py

```diff
@@ -78,12 +78,19 @@
                         help='The name of the requested chain',
                         choices=chain_dict.keys(),
                         )
+    parser.add_argument('--type',
```

Comment:
Rename to be more meaningful, e.g. '--promotions-file-type'.
```suggestion
    parser.add_argument('--promotions-file-type',
```